### ISSUE TITLE
Dataset of spline and edge points not the same

### ISSUE THREAD

**Issue by maxijsseldijk** [ShaoKAi100812/Pendant_Drop_Tensiometry_Prediction#2]
For Physicsnet the datasets should be the same for both edge points and spline with the only difference that the spline should use the spline coefficients. However I tried by implementing it directly in the Datasplit class using the same subsetSampler. However, the DataLoader object keeps printing out different values due to the random sampler which means that it is difficult to get the same test image to compare the performance

 self.train_sampler = SubsetRandomSampler(self.train_indices)

self.val_sampler = SubsetRandomSampler(self.val_indices)

self.test_sampler = SubsetRandomSampler(self.test_indices)`

Is used for both datasets:

@lru_cache(maxsize=4)
    def get_train_loader(self, batch_size=64, num_workers=0):
        logging.debug('Initializing train dataloader')
      self.train_loader = DataLoader(self.dataset, batch_size=batch_size, sampler=self.train_sampler, shuffle=False, num_workers=num_workers)
        self.train_loader_test = DataLoader(self.dataset_test, batch_size=batch_size, sampler=self.train_sampler, shuffle=False, num_workers=num_workers)
        return self.train_loader, self.train_loader_test

Which is then output using:

split_points = DataSplit(my_data_set_points,my_data_set_spline, shuffle=True)

train_loader_points, val_loader_points, test_loader_points, train_loader_spline, val_loader_spline, test_loader_spline = split_points.get_split(batch_size=64, num_workers=0)

**Comment by ShaoKAi100812** [comment]
My idea is to try training the two models first and then apply the evaluation by generating the new data without shuffle, so we can  evaluate the result through the same subset from PictureNet to PhysicsNet. 
However, we will first evaluate the PictureNet prediction of rr and zz based on two methods (point position and spline inference) individually. Then, PhysicsNet.

**Comment by ShaoKAi100812** [comment]
I saw u modify the DatasSplit class, which I think is not very necessary.
I will open a remote branch n push my modification over there first, in case the conflict occurs.

**Comment by maxijsseldijk** [comment]
Yes that is the second option indeed to generate new data for the evaluation of the model.  However, my idea was to already implement that in the datasplit in order to also evaluate Physicsnet with the same test data. But this does not work due to the sampler also picking random data points within a certain range. I think it would be better to generate new data in MATLAB and then shuffle it there so we can use it to evaluate PictureNet
